src/vignette/util/image_response.clj

```diff
@@ -49,9 +50,10 @@
 
 (defn add-content-disposition-header
   [response-map image-map]
-  (header response-map "Content-Disposition"
-          (format "inline; filename=\"%s\""
-                  (original image-map))))
+  (let [requested-format (query-opt image-map :format)
+        filename (cond-> (original image-map)
+                         requested-format (str "." requested-format))]
```

Comment:
Yeah. Seems easier than trying to strip out the old extension, especially since it might not be there.